src/xds/client.rs

```diff
@@ -304,13 +304,16 @@ impl AdsClient {
         let ns = ns.as_deref().unwrap_or("");
         let node = std::env::var("NODE_NAME");
         let node = node.as_deref().unwrap_or("");
+        let ambient_type = std::env::var("AMBIENT_TYPE");
+        let ambient_type = ambient_type.as_deref().unwrap_or("");
         Node {
             id: format!("sidecar~{ip}~{pod_name}.{ns}~{ns}.svc.cluster.local"),
             metadata: Some(Self::build_struct([
                 ("POD_NAME", pod_name),
                 ("POD_NAMESPACE", ns),
                 ("INSTANCE_IP", ip),
                 ("NODE_NAME", node),
+                ("AMBIENT_TYPE", ambient_type),
```

Comment:
maybe should just be a constant? ztunnel

Reply:
or I can add a default value on the env var lookup similar to what is done for `INSTANCE_IP` above.

Reply:
I had a PR in the ztunnel charts to include the env var, but I can also just add the default value: https://github.com/istio/istio/pull/42527

Reply:
I don't think there is any use case to not have it hardcoded. I think we should do https://github.com/istio/istio/issues/42313 actually, eventually

Reply:
Is there any chance in the future a "sandwich" ztunnel might need different config?

Reply:
Maybe but we can probably define that as needed?